### PR TITLE
Ncgames 7-POST /api/review/:review_id/comment

### PR DESCRIPTION
**Pull request by FisherTom**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -146,6 +146,70 @@ describe("GET requests", () => {
...
+            body: "Test Comment",
+            comment_id: expect.any(Number),
+            created_at: expect.any(String),
+            review_id: 1,
+            votes: 0,
+          });
+        });
+    });
+    test('400: "Bad request" if given bad review id', () => {
+      const testComment = {
```
Comment (NRMSMITH) [inline __tests__/app.test.js:173]: change 'bad review id' to 'invalid'

File: __tests__/app.test.js [hunk 2]
```diff
@@ -146,6 +146,70 @@ describe("GET requests", () => {
...
+        username: "not_a_username",
+        body: "Test Comment",
+      };
+      return request(app)
+        .post("/api/reviews/10000/comments")
+        .send(testComment)
+        .expect(400)
+        .then((response) => {
+          expect(response.body.msg).toBe("Bad request");
+        });
```
Comment (NRMSMITH) [inline __tests__/app.test.js:209]: Shouldn't this be a 404? "not_a_username" technically could be a username at somepoint, it just isn't one that is in the testdata.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -146,6 +146,70 @@ describe("GET requests", () => {
...
+    });
```
Comment (NRMSMITH) [inline __tests__/app.test.js:197]: There is a comment body in this one - so this should be '404: review not found'; not a missing comment body.

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Requesting changes - you've muddled up your tests a little bit here. 404 means that something could exist, but it hasn't been found. 400 means that there has been an error on the client's end (they've used the wrong data type for id etc)

File: __tests__/app.test.js [hunk 4]
```diff
@@ -146,6 +146,91 @@ describe("GET requests", () => {
...
+        body: "Test Comment",
+      };
+
+      return request(app)
+        .post("/api/reviews/1/comments")
+        .send(testComment)
+        .expect(201)
+        .then((response) => {
+          expect(response.body).toMatchObject({
+            author: "dav3rid",
```
Comment (Rosem10) [inline __tests__/app.test.js:163]: Please also check that this has actually made it's way into the database (at the moment we're getting the object back, but there's no verification that this change has been comitted to the database). Once you've done the patch, utilise your previous getCommentsByArticleId to check that your new comment is added.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -146,6 +146,91 @@ describe("GET requests", () => {
...
+        .then((response) => {
+          expect(response.body.msg).toBe("Not found");
+        });
+    });
+    test('404: "Not found" if given nonexistant username', () => {
+      const testComment = {
+        username: "not_a_username",
+        body: "Test Comment",
+      };
+      return request(app)
```
Comment (Rosem10) [inline __tests__/app.test.js:212]: Nice - but also what about a review Id of the wrong type?

File: app/error-handlers.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
+// app.all("/*", (request, response) => {
```
Comment (Rosem10) [inline app/error-handlers.js:1]: Remove commented code, please!

**Review by Rosem10 — APPROVED** [review]
Couple of little bits to add please!

**Review by Rosem10 — APPROVED** [review]
Really nice! (although those comments still need to be removed before your next PR please!)

**Pull request merged**